doc/api/test.md

```diff
@@ -476,8 +476,10 @@ all tests have completed. If the [`NODE_V8_COVERAGE`][] environment variable is
 used to specify a code coverage directory, the generated V8 coverage files are
 written to that directory. Node.js core modules and files within
 `node_modules/` directories are, by default, not included in the coverage report.
-However, they can be explicitly included via the [`--test-coverage-include`][] flag. If
-coverage is enabled, the coverage report is sent to any [test reporters][] via
+However, they can be explicitly included via the [`--test-coverage-include`][] flag.
+By default all the matching test files are excluded from the coverage report.
+Exclusions can be overridden by using the [`--test-coverage-exclude`][] flag.
+If coverage is enabled, the coverage report is sent to any [test reporters][] via
```

Comment:
If I specify `--test-coverage-exclude='**/*.ts'` and the test files are `**/*.test.ts`, would that inadvertently force them to get included? (If so, this feature seems to effectively do nothing)

Some examples would be good :)

Reply:
Yes, it will!  

As soon as you define an exclusion, you're overriding the default values.  
This is the same behavior as the exclusion flag in `c8`, if I'm not mistaken.  
If we simply append the tests to the exclusions, the user wouldn't have any way to add them back.  

The important point is that this isn't `inadvertent`; it's the expected behavior.  

The default values are something like `[defaultPatterns]` and get overridden as soon as the user provides at least one value.

Do you have any suggestions on how to make this clearer to the user?
Does this behaviour make sense to you? 😁

Reply:
If that's the behaviour, I think this will unfortunately be useless in most cases :(

Reply:
I agree, at least until we have a configuration file that allows the user to do something like:
<img width="804" alt="image" src="https://github.com/user-attachments/assets/9233935c-329f-4b8b-a788-152f2d573089" />
https://vitest.dev/config/#coverage-exclude